src/js/login/reducers/login.js

```diff
@@ -35,13 +37,13 @@ function loginStuff(state = initialState, action) {
       return _.set('currentlyLoggedIn', action.value, state);
 
     case UPDATE_LOGIN_URL:
-      return _.set('loginUrl', action.value, state);
+      return _.set('loginUrl', appendQuery(action.value, { clientId: gaClientId() }), state);
```

Comment:
I think your check is fine. I'd go with:
function getLoginURL(default_url) {
 let clientId = gaClientId()
 return (clientId === undefined) ? default_url : appendQuery(default_url, { clientId: clientId })
} 
...
return _.set('loginUrl', getLoginURL(action.value) , state)

But that's basically just out of my own ignorance of how JS and the appendQuery library would handle the `undefined` return in the case GA isn't there. The proverbial abundance of caution approach.

Reply:
I can confirm that when you try to add a query param of value `undefined`, `append-query` ignores that param.  E.g:
```
appendQuery('http://foo', { bar: undefined })
```
will result as 
```
http://foo
```